api/src/test/java/org/openmrs/module/ohricore/aop/EncounterInterceptAdviceTest.java

```diff
@@ -0,0 +1,103 @@
+package org.openmrs.module.ohricore.aop;
+
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.openmrs.Encounter;
+import org.openmrs.api.AdministrationService;
+import org.openmrs.api.EncounterService;
+import org.openmrs.api.context.Context;
+import org.openmrs.api.context.UserContext;
+import org.openmrs.module.ohricore.engine.ConceptComputeTrigger;
+import org.openmrs.test.jupiter.BaseContextSensitiveTest;
+import org.openmrs.test.jupiter.BaseModuleContextSensitiveTest;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Autowired;
+import static org.powermock.api.mockito.PowerMockito.mockStatic;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.lang.reflect.Method;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+/**
+ * @author MayanjaXL, Amos, Stephen, smallGod date: 18/06/2021
+ */
+//TODO: Request Eudson to create upstream branch: "feature/ohri-146-patient-state-tests"
+@RunWith(PowerMockRunner.class)
+@PrepareForTest(Context.class)
+public class EncounterInterceptAdviceTest extends BaseModuleContextSensitiveTest {
+	
+	protected static final String ENCOUNTER_DATA_XML = "org/openmrs/module/ohricore/include/BasicEncounter-initialData.xml";
+	
+	protected static final String CONCEPTS_DATA_XML = "org/openmrs/module/ohricore/include/HIVComputedConcept-initConceptData.xml";
+	
+	protected static final String TEST_ENCOUNTER_UUID = "430bbb70-6a9c-4e1e-badb-9d1034b1b5e9";
+	
+	private Method methodInvoked;
+	
+	@Autowired
+	EncounterService encounterService;
+	
+	@Before
+	public void setUpEncounter() throws ClassNotFoundException, NoSuchMethodException {
+		
+		mockStatic(Context.class);
+		EncounterService encounterService = mock(EncounterService.class);
+		when(Context.getEncounterService()).thenReturn(encounterService);
+		
+		executeDataSet(ENCOUNTER_DATA_XML);
+		executeDataSet(CONCEPTS_DATA_XML);
+		
+		Class clazz = Class.forName("org.openmrs.api.impl.EncounterServiceImpl");
+		methodInvoked = clazz.getDeclaredMethod(ConceptComputeTrigger.SAVE_ENCOUNTER, Encounter.class);
+	}
+	
+	@Test
+	public void afterReturning_shouldCompute() throws Throwable {
+		
+		Encounter encounter = encounterService.getEncounterByUuid(TEST_ENCOUNTER_UUID);
+		System.out.println("encounter: " + encounter.getUuid());
+		assertNotNull(encounter, "Test Encounter should be available");
+		assertNotNull(methodInvoked);
+		assertEquals("saveEncounter", methodInvoked.getName());
+		new EncounterInterceptorAdvice().afterReturning(null, methodInvoked, new Object[] { encounter }, null);
+	}
+}
+
+//public class EncounterInterceptAdviceTest extends BaseContextSensitiveTest {
```

Comment:
@smallgod Delete all unused code, rather than commenting it out

Reply:
@ssmusoke this is done!